olp-cpp-sdk-core/include/olp/core/porting/shared_mutex.h

```diff
@@ -32,6 +32,15 @@
 #define HAVE_PTHREAD_RWLOCK
 #endif
 
+#ifndef THROW_OR_ABORT
+# if __cpp_exceptions
+#  define THROW_OR_ABORT(exception) (throw (exception))
+# else
+#  include <cstdlib>
+#  define THROW_OR_ABORT(exception) (std::abort())
```

Comment:
This will result in compilation errors or warnings for unused vars. Please add cast to void for the exception class.

```cpp
(void)exception;
std::abort();
```

Reply:
How this can cause any errors or warnings? I don't understand. AFAIK preprocessor just substitutes throw_or_abort(...) with std::abort() without any object creation or something else. We still on the parse stage.

Reply:
The exception object is created and it is a class but not used anywhere if you only use std::abort().